controllers/sriovnetworknodepolicy_controller.go

```diff
@@ -584,10 +584,7 @@ func createDevicePluginResource(
 	if len(p.Spec.NicSelector.RootDevices) > 0 {
 		netDeviceSelectors.RootDevices = append(netDeviceSelectors.RootDevices, p.Spec.NicSelector.RootDevices...)
 	}
-	// Removed driver constraint for "netdevice" DeviceType
```

Comment:
Hi I think this change is not right.

can you add a unit test to test partitional device?
I think this change will break the case where we have one PF that we split in 2 
10 vfio-pci (resource name: vfio)
10 netdevice (resource name: netdev)

I think if we don't add the driver the device plugin will select all of the VFs to the vfio pool

Reply:
I added [2cbd5b5](https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/923/commits/2cbd5b5d8e4e81d8bacde16425bb4f299e686495).

The gist is that the scenario with vfio-pci and netdevice must go with pf partitioning, using the `<pfname>#<vfindexes>` syntax. The vf indexes are enough to select the right VFs. 

The Drivers selector, on the other hand, is not accurate enough (what happens if I have two vfio-pci resources?), so it can be dropped, as it is redundant.